file.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"fmt"
+	"net/http"
+	"io"
+	"path/filepath"
+	"bytes"
+	"archive/zip"
+	"strings"
+)
+
+// Download the zip file at the given URL to a temporary local directory.
+// Returns the asbolute path to the downloaded zip file.
+// IMPORTANT: You must call defer os.RemoveAll(dir) when done with this directory!
+func downloadGithubZipFile(repoOwner, repoName, gitTag, githubToken string) (string, *fetchError) {
+	// Create a temp directory
+	tempDir, err := ioutil.TempDir("", "")
```

Comment:
I was sure this was a bug and had to look it up in the Go API. `ioutil.TempDir` is just awful. You pass an *empty string* string as the first argument to indicate you want to use the default temporary directory? And if you pass an *empty string* for the second argument, it *seems* to indicate that it'll generate something random to avoid conflicts? WTF were the Go designers thinking?

For future maintainers, it might be worth explaining how this horrid API works in a comment :)